moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryKnackDataTrackerSync.js

```diff
@@ -25,38 +25,108 @@ const buildProjectUrl = (scene, view, knackProjectId) => {
  * @returns integer
  */
-const getProjectSignals = project => {
-  const allProjectFeatures = project.moped_proj_components
-    .map(projectComponent => {
-      // for each component, extract feature geojson from each project feature (moped_proj_features.feature)
-      // every feature must have a feature.feature jsonb, but we'll nullish chain just to be safe
-      return projectComponent.moped_proj_features.map(
-        feature => feature?.feature
-      );
-    })
-    // then flatten this array of feature arrays
-    .flat()
-    // and filter any undefined components, which should never happen
-    .filter(feature => feature);
-  // now filter our features for only those with a `signal_id` property
-  const allProjectSignals = allProjectFeatures.filter(
-    feature => feature.properties?.signal_id
+const useProjectSignals = (project) =>
+  useMemo(
+    () =>
+      project.moped_proj_components
+        .map((component) => component.feature_signals)
+        .flat()
+        .filter((signal) => !!signal),
+    [project]
   );
-  // and shape the object we need to render links and such
-  return allProjectSignals.map(signal => {
-    return {
-      signal_id: signal.properties.signal_id,
-      knack_id: signal.properties.id,
-    };
+
+/**
+ * Object to hold headers which need to be sent as part of the a Knack API call
+ */
+const REQUEST_HEADERS = {
+  "Content-Type": "application/json",
+  "X-Knack-Application-Id": process.env.REACT_APP_KNACK_DATA_TRACKER_APP_ID,
+  "X-Knack-REST-API-Key": "knack",
+};
+
+const FIELD_MAP = {
+  production: {
+    project_id: "field_4133",
+    project_name: "field_3857",
+    current_phase_name: "field_4136",
+    signals_connection: "field_3861",
+    moped_url_object: "field_4162",
+  },
+  other: {
+    project_id: "field_4133",
+    project_name: "field_3857",
+    current_phase_name: "field_4136",
+    signals_connection: "field_3861",
+    moped_url_object: "field_4162",
+  },
+};
+
+const getUrlObject = (project) => {
+  const url =
+    process.env.REACT_APP_KNACK_DATA_TRACKER_URL_BASE + project.project_id;
+  return {
+    url: url,
+    label: project.project_name,
+  };
+};
+
+/**
+ * Exract project's current phase name from projet data - assumes the 0-index project phase
+ * is the project's current phase
+ * @param {object} project - project as returned from our SUMMARY_QUERY
+ * @returns
+ */
+const getCurrentPhase = (project) =>
+  project.moped_proj_phases?.[0].moped_phase.phase_name || null;
+
+/**
+ * Function to build up a JSON object of the fields which need to be updated in a call to Knack. This is needed
+ * because if you update the project number field, even with the same, extant number, Knack returns an error.
+ * @param {object} project - project data as returned from our SUMMARY_QUERY
+ * @param {[object]} signals - array of from the feature_signals table
+ * @returns string
+ */
+const buildBody = (project, signals) => {
```

Comment:
when i was editing this code i realized that my removal of `moped_project.current_status` ([#871](https://github.com/cityofaustin/atd-moped/pull/871)) broke part of our knack payload! we were using `current_status` here, but i missed it b/c it was we were parsing that column name out of the env var. 

anyway—i opted to move those knack field maps to a hard-coded object in this component. it's not pretty but i hopefully a lil clearer what data we depend on here.

Reply:
otherwise the changes to this function are really just extracting it from the main component body.

Reply:
Love that you are leaving this code in a better place than you found it! 🙏